src/arbitrage.py

```diff
@@ -0,0 +1,82 @@
+class Query (object):
+
+    def __init__(self, limit=0, stock_names=None, api=None):
+        self.limit = limit
+        self.names = stock_names
+        self.api = api
+
+    def pull(self, stock_name):
+        """Connects to API to pull data for a given stock and returns a StockData object."""
+        return self.api.get(stock_name)
+
+    def check(self, expiry_dates=None, option_prices=None, underlying_cost=None):
+        """Checks to see if there exists an expiry date for which
+        the arbitrage formula for options holds"""
+        result = False
+        for expiry in expiry_dates:
+            diff_cost = abs(option_prices[expiry][0] - option_prices[expiry][1])
+            if diff_cost / underlying_cost > self.limit:
+                result = True
+        return result
+
+    def find_opportunities(self):
+        """Prints the stocks that satisfy the arbitrage opportunity requirements"""
+        for name in self.names:
+            data = self.pull(name)
+            expiry_dates = data.get_dates()
+            underlying_cost = data.get_underlying()
+            option_prices = data.get_option_prices()
+            if self.check(expiry_dates, option_prices, underlying_cost):
+                print(name)
+
+class HistoricalQuery(Query):
+    def __init__(self, timeframe, limit=0, stock_names = None, api = None):
+        super().__init__(limit, stock_names, api)
+        self.timeframe = timeframe
+    
+    def pull(self, stock, time):
+        """Connects to a API to get historical data"""
+        return
+
+    def find_opportunities(self):
+        for i in range(len(self.timeframe)):
+            for j in range(i, len(self.timeframe)):
+                for name in self.names:
+                    underlying_time = self.timeframe[i]
+                    option_time = self.timeframe[j]
+                    underlying_cost = self.pull(name, underlying_time).get_underlying()
+                    option_data = self.pull(name, option_time)
+                    expiry_dates = option_data.get_dates()
+                    option_prices = option_data.get_option_prices()
+                    if self.check(expiry_dates, option_prices, underlying_cost):
+                        print(name, underlying_time, option_time)
+            
+class StockData(object):
```

Comment:
What is this object? It does not seemed currently used.

I don't know if you should have both StockData and API as a wrapper around real stock data. What is the purpose of each one?

Reply:
API can get many different stocks but StockData is only supposed to represent one stock. Also want to use to differentiate between historical data and live